pyxem/tests/signals/test_polar_diffraction2d.py

```diff
@@ -142,6 +143,46 @@ def test_power_inplace(self, flat_pattern):
         assert ac is None
         assert isinstance(flat_pattern, Power2D)
 
+class TestPearsonCorrelation:
+    @pytest.fixture
+    def flat_pattern(self):
+        pd = PolarDiffraction2D(data=np.ones(shape=(2, 2, 5, 5)))
+        pd.axes_manager.signal_axes[0].scale = 0.5
+        pd.axes_manager.signal_axes[0].name = "theta"
+        pd.axes_manager.signal_axes[1].scale = 2
+        pd.axes_manager.signal_axes[1].name = "k"
+        return pd
+
+    @pytest.mark.parametrize("kmin", [0, 1])
+    @pytest.mark.parametrize("kmax", [4, 5])
+    def test_pcorrelation_signal(self, flat_pattern, kmin, kmax):
+        rho_w = flat_pattern.get_pearson_correlation()
+        assert isinstance(rho_w, Symmetry1D)
+        rho = flat_pattern.get_pearson_correlation(kmin=kmin, kmax=kmax)
+        assert isinstance(rho, Symmetry1D)
```

Comment:
This test (if written slightly differently) would have caught the bug mentioned above. Rather than running the default case you can instead have:

```python
@pytest.mark.parametrize("kmin", [0, 1,None])
@pytest.mark.parametrize("kmax", [4, 5,None])
def test_pcorrelation_signal(self, flat_pattern, kmin, kmax):
        rho = flat_pattern.get_pearson_correlation(kmin=kmin,kmax=kmax)
        assert isinstance(rho_w, Symmetry1D)
```